dev-tools/mage/pkgtypes.go

```diff
@@ -904,15 +905,24 @@ func addFileToZip(ar *zip.Writer, baseDir string, pkgFile PackageFile) error {
 
 // addFileToTar adds a file (or directory) to a tar archive.
 func addFileToTar(ar *tar.Writer, baseDir string, pkgFile PackageFile) error {
+	excludedFiles := []string{
+		"cloud-defend",
+		"cloud-defend.spec.yml",
+	}
+
 	return filepath.Walk(pkgFile.Source, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
 			if pkgFile.SkipOnMissing && os.IsNotExist(err) {
 				return nil
 			}
-
 			return err
 		}
 
+		if slices.Contains(excludedFiles, info.Name()) {
+			// it's a file we have to exclude
+			return nil
```

Comment:
We should log what we are ignoring so we don't forget this is happening.

Reply:
Done in 5d67bf4e97e7b7f8d32cdfd8fff93dc6294e9b5d

Reply:
This seems like a hack.

I think it would be better to define this higher up so its clean and clearly defined why its included and not. At this level it just seems like a hack.

Reply:
Actually this is a hack: the best solution would be to be able to have a clear way to define package for docker builds compare to other builds but this would require a way bigger change.

Reply:
This is definitely a hack, but it's a small hack that helps solve a customer problem.

I'm going to approve this but the hack version of this can't live for very long.

Reply:
This is a hack but it is a quick win to reduce the package size for linux... I filed a new issue to replace this change with something more structured https://github.com/elastic/elastic-agent/issues/4588